aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -142,6 +148,30 @@ struct TORCH_API DispatchKeyExtractor final {
     dispatch_arg_indices_reverse_ = c10::utils::bitset();
   }
 
+  DispatchKeySet getDispatchKeySet(DispatchKeySet ks) const {
+    // Keys that are fallthrough should be skipped
+    if (requiresBitsetPerBackend_) {
+      auto backend_idx = ks.getBackendIndex();
+      ks = impl::computeDispatchKeySet(
+          ks, nonFallthroughKeysPerBackend_[backend_idx]);
+    } else {
+      ks = impl::computeDispatchKeySet(ks, nonFallthroughKeys_);
+    }
+
+    // If the Functionality Key located on the leftmost side of ks has a
+    // corresponding callback function, it means that there is additional
+    // calculation for the functionality Key.
+    // Currently only used by Autocast, as AutocastFunctionality acts as
+    // a Per Backend Functionality key, which is not sufficient to determine
+    // which backend is enabled.
+    auto found = getDKFuncMaps().find(ks.highestFunctionalityKey());
+    if (found != getDKFuncMaps().end()) {
+      ks = found->second(ks);
```

Comment:
I think my first ask is to not use a dynamic dispatch mechanism here. We should just have the logic inline here, where hopefully it can be branch predicted much more easily.

Reply:
Actually, this is the same thing I asked about months ago. Can you talk me through this part?